src/verifier/core/authorizing.py

```diff
@@ -131,7 +130,9 @@ def processPresentations(self):
             cred_state = None
             if state.state == AUTH_EXPIRE and age > datetime.timedelta(seconds=self.TimeoutAuth*2):
                 self.vdb.iss.rem(keys=(aid,))
-            elif state.state != AUTH_EXPIRE and age > datetime.timedelta(seconds=self.TimeoutAuth):
+            # We keep revoked credentials in the DB because their auth should never expire and the state
+            # must always be AUTH_REVOKED to avoid logging in again with the older version of the credential
+            elif state.state != AUTH_REVOKED and state.state != AUTH_EXPIRE and age > datetime.timedelta(seconds=self.TimeoutAuth):
```

Comment:
Do you think we need an additional elif for AUTH_SUCCESS, where we process_revocation again, just like we did in the 'verifying' stage.... the reason is that we will eventually have a watcher that is monitoring the credential status regularly. If the AUTH_SUCCESS occurred and then later the issuer revokes the credential, we need to update our status to AUTH_REVOKED. I know we don't have the watcher now.... and now that i'm thinking about it, perhaps if the watcher sees an update to revoked... then we'll process that automatically as a new presentation... hm, yea i think maybe that is better but what do you think @aydarng . If its a new presentation, can you just test a DIFFERERNT user (such as the issuer) presenting the revoked credential and verify that the issuee account is removed.

Reply:
Sure, I'll test that.

Reply:
Great, let me know if its successful i'll approve it.